source/agora/consensus/SlashPolicy.d

```diff
@@ -105,7 +110,7 @@ public class SlashPolicy
 
         foreach (idx, utxo_key; keys)
         {
-            if (!this.hasRevealedPreimage(height, utxo_key))
+            if (!this.hasRevealedPreimage(this.heightForSlashing(height), utxo_key))
```

Comment:
Am I understanding this correctly? This slashes the node if it doesn't reveal at least X images in advance, where X is the threshold?

Reply:
The threshold, X is to specify how many blocks a validator can miss pre-images before being slashed. If it's set to 0 which is the default value, the validator is slashed as soon as the case of missing pre-images is detected. If 1 is set to the value, the validator can miss the pre-images for one block. This concept is based on the cases of missing preimages, which is describe [the original design of the slashing protocol](https://github.com/bpfkorea/agora/issues/1076).